config.py

```diff
@@ -2763,7 +2775,14 @@ def get_decoding_contrasts() -> Iterable[Tuple[str, str]]:
 def failsafe_run(
     on_error: OnErrorT,
     script_path: PathLike,
+    get_input_fnames: Optional[Callable] = None,
 ):
+    if memory_location is None or get_input_fnames is None:
+        # No caching is needed
+        memory = Memory(location=None)  # no op
```

Comment:
FYI @agramfort you had this as `location='.'` which was not a no-op. I assume this was an oversight and not intentional... that, or I don't understand `Memory` or its docstring properly.

But I left the default of `memory_location='.'` in the config for now, which means by default after this PR users will see benefits and all caching will probably be in the `mne-bids-pipeline` directory

Reply:
We should dump the cache in the derivatives directory, don't you agree?

Reply:
Not sure which is better, but you're probably right. Maybe we alias `memory_location = True` to mean "in the derivatives folder"? I don't want to use a string default because the user might actually want that specific path (relative to `mne-bids-pipeline`)

Reply:
This has been implemented as:
```
If not None (or False), caching will be enabled and the cache files will be
stored in the given directory. The default (True) will use a
``'.joblib_cache'`` subdirectory in the BIDS derivative root of the dataset.
```